cmd/tapa/main.go

```diff
@@ -140,7 +140,7 @@ func main() {
 			vfiomech.MECHANISM:   chain.NewNetworkServiceClient(vfio.NewClient()),
 			kernelmech.MECHANISM: chain.NewNetworkServiceClient(kernel.NewClient()),
 		}),
-		interfacename.NewClient("nsc", &interfacename.RandomGenerator{}),
+		interfacename.NewClient("nsm-", &interfacename.CounterGenerator{}),
```

Comment:
> Before, the interface name was set only to "nsc", so if 2 interfaces were requested, then only 1 was created. The interfacename chain element is required, kernel.NewClient has a bug if 2 connections have too identical id + NS name. Now interfaces are in the format: nsm-<number>

The reason for "bug" with kernel.NewClient is that besides the ns name it relies on connection.Id to generate the interface name. Yet, connection.Id in TAPA does not have "random" prefix at its start, unlike the fullmesh client in proxy (which prepends the id to be used with uuid.New().String() ).

I also remember, that this intefaceName chain component was way too immature. For example it was generating new ids even if it wasn't necessary. I slightly remember it leading to problems, but can't recall the details. Maybe it was mostly related to the proxy when testing lb failovers or such.

On the other hand, letting nsm name the interfaces is not bulletproof either, mostly because of the fixed size of the interface name. Meaning that in case of "longer" ns names only the first 4 characters of the random string could be used to generate the interface name.

Reply:
right, the NSM generator is using the connection id, and in this multi-conduit case, the nsm interfaces will be the same if the conduit names are too similar (since the beginning of the connection id will be the conduit name). I also got the request the names should be in this format: nsm-<number>.
I think I also fixed the generation of new interface ids even if it wasn't necessary in [pkg/nsm/interfacename/common.go](https://github.com/Nordix/Meridio/pull/239/files#diff-b10cf177849d9666f321d3d2106f98085378b351f277314eae456f38df4a4a3f)

Reply:
I see, I will run some tests then to see how the name generation performs.

Reply:
this should be fixed now